conda_build/main_build.py

```diff
@@ -308,6 +309,8 @@ def execute(args, parser):
                     (conda_version[lang], version))
 
     if args.skip_existing:
+        if not isdir(config.bldpkgs_dir):
+            makedirs(config.bldpkgs_dir)
```

Comment:
@asmeurer Moving `makedirs` into `config` will look awkward. I would probably a several changes:
1. add a new function something along the lines `update_local_index` that will check `noarch` and `$PLATFORM` build indecies.
2. update conda's fetch_repodata so it is not hardcoded against `'noarch'`, but rather has an option to silently (but with debug logging) skip non-existent repos, and in the worst case log a warning instead of rasing an exception if a given repo is unreachable.